pynbody/snapshot/pkdgravhdf.py

```diff
@@ -0,0 +1,153 @@
+import configparser
+import logging
+import warnings
+
+from .. import config_parser, family, units
+from .gadgethdf import GadgetHDFSnap, _GadgetHdfMultiFileManager
+
+logger = logging.getLogger('pynbody.snapshot.pkdgravhdf')
+
+try:
+    import h5py
+except ImportError:
+    h5py = None
+
+_pkd_default_type_map = {}
+for x in family.family_names():
+    try:
+        _pkd_default_type_map[family.get_family(x)] = \
+            [q.strip() for q in config_parser.get('pkdgrav3hdf-type-mapping', x).split(",")]
+    except configparser.NoOptionError:
+        pass
+
+_pkd_all_hdf_particle_groups = []
+for hdf_groups in _pkd_default_type_map.values():
+    for hdf_group in hdf_groups:
+        _pkd_all_hdf_particle_groups.append(hdf_group)
+
+
+class _PkdgravHdfMultiFileManager(_GadgetHdfMultiFileManager) :
+    _nfiles_groupname = "Header"
+    _nfiles_attrname = "NumFilesPerSnapshot"
+    _namemapper_config_section = "pkdgrav3hdf-name-mapping"
+
+    def __init__(self, filename, mode='r') :
```

Comment:
This appears to be largely copy/pasted from _GadgetHdfMultiFileManager, which obscures what is supposed to be different/the same.

As far as I can see it's only the filename that is different.

As such, could you change `_GadgetHdfMultiFileManager` so that the filename is made in a `_make_filename_for_cpu` method, and then you only need to override that one method?